pkg/chains/formats/slsa/v2alpha2/internal/resolved_dependencies/resolved_dependencies.go

```diff
@@ -0,0 +1,361 @@
+
+			var url string
+			var revision string
+			for _, param := range input.ResourceSpec.Params {
+				if param.Name == "url" {
+					url = param.Value
+				}
+				if param.Name == "revision" {
+					revision = param.Value
+				}
+			}
+			rd.URI = attest.SPDXGit(url, revision)
+			*resolvedDependencies = append(*resolvedDependencies, rd)
+		}
+	}
+}
+
+func addPipelineTask(logger *zap.SugaredLogger, pro *objects.PipelineRunObject, resolvedDependencies *[]v1.ResourceDescriptor) error {
+	pSpec := pro.Status.PipelineSpec
+	if pSpec != nil {
+		pipelineTasks := append(pSpec.Tasks, pSpec.Finally...)
+		for _, t := range pipelineTasks {
+			tr := pro.GetTaskRunFromTask(t.Name)
+			// Ignore Tasks that did not execute during the PipelineRun.
+			if tr == nil || tr.Status.CompletionTime == nil {
+				logger.Infof("taskrun status not found for task %s", t.Name)
+				continue
+			}
+			// add remote task configsource information in materials
+			if tr.Status.Provenance != nil && tr.Status.Provenance.RefSource != nil {
+				rd := v1.ResourceDescriptor{
+					Name:   pipelineTaskConfigName,
+					URI:    tr.Status.Provenance.RefSource.URI,
+					Digest: tr.Status.Provenance.RefSource.Digest,
+				}
+				*resolvedDependencies = append(*resolvedDependencies, rd)
+			}
+			// add step images
+			if err := addStepImagesToResolvedDependencies(tr.Status.Steps, resolvedDependencies); err != nil {
+				return err
+			}
+
+			// add sidecar images
+			if err := addSidecarImagesToResolvedDependencies(tr.Status.Sidecars, resolvedDependencies); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func addPipelineResults(ctx context.Context, pro *objects.PipelineRunObject, resolvedDependencies *[]v1.ResourceDescriptor) {
+	var commit, url string
+	// search spec.params
+	for _, p := range pro.Spec.Params {
+		if p.Name == attest.CommitParam {
+			commit = p.Value.StringVal
+			continue
+		}
+		if p.Name == attest.URLParam {
+			url = p.Value.StringVal
+		}
+	}
+
+	// search status.PipelineSpec.params
+	if pro.Status.PipelineSpec != nil {
+		for _, p := range pro.Status.PipelineSpec.Params {
+			if p.Default == nil {
+				continue
+			}
+			if p.Name == attest.CommitParam {
+				commit = p.Default.StringVal
+				continue
+			}
+			if p.Name == attest.URLParam {
+				url = p.Default.StringVal
+			}
+		}
+	}
+
+	// search status.PipelineRunResults
+	for _, r := range pro.Status.PipelineResults {
+		if r.Name == attest.CommitParam {
+			commit = r.Value.StringVal
+		}
+		if r.Name == attest.URLParam {
+			url = r.Value.StringVal
+		}
+	}
+	if len(commit) > 0 && len(url) > 0 {
+		url = attest.SPDXGit(url, "")
+		*resolvedDependencies = append(*resolvedDependencies, v1.ResourceDescriptor{
+			Name:   inputResultName,
+			URI:    url,
+			Digest: map[string]string{"sha1": commit},
+		})
+	}
+
+	sms := artifacts.RetrieveResolvedDependenciesFromStructuredResults(ctx, pro, artifacts.ArtifactsInputsResultName)
+	for i := range sms {
+		sms[i].Name = inputResultName
+	}
```

Comment:
Do you think it would be a good idea to pass the name to that helper function instead of using another for loop to set the name? This way, we will have a single place to set that type of resolvedDependencies.

Reply:
Oh yeah, we can do that.